src/server/status/index.js

```diff
@@ -18,7 +19,11 @@ export default function (kbnServer, server, config) {
   server.route(wrapAuth({
     method: 'GET',
     path: '/api/status',
-    handler: function (request, reply) {
+    handler: async function (request, reply) {
+      const stats = getStats(
+        server.config().get('kibana.index'),
+        request,
+        server.plugins.elasticsearch.getCluster('admin').callWithRequest);
```

Comment:
I think we'll want the API endpoint to still use callWithRequest, and we could make the class interface for fetching optionally use an admin client.  Would that work @tsullivan?